sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -444,138 +441,98 @@
     }
 
     /**
-     * Set the {@link ConfigurationSetting} to read-only with a matching {@code key}, and optional {@code label}
-     * combination.
+     * Sets the read-only status for the {@link ConfigurationSetting}. If {@code isReadOnly} is {@code true} the
+     * configuration setting will be read-only, {@code false} will clear the read-only status.and optional
+     * {@code label} combination. If the {@code isReadOnly} is true or null, the setting will be set to read-only.
+     * If false, the setting will not be read-only.
```

Comment:
Let's simplify this, this is the leading paragraph and should simply state what this API is used for, `Sets the read-only status for the {@link ConfigurationSetting}.`